client/src/thin_client.rs

```diff
@@ -142,18 +141,14 @@ impl ThinClient {
         timeout: Duration,
     ) -> Self {
         let rpc_client = RpcClient::new_socket_with_timeout(rpc_addr, timeout);
-        Self::new_from_client(tpu_addr, transactions_socket, rpc_client)
+        let tpu_connection = C::new(transactions_socket, tpu_addr);
+        Self::new_from_clients(rpc_client, tpu_connection)
     }
 
-    fn new_from_client(
-        tpu_addr: SocketAddr,
-        transactions_socket: UdpSocket,
-        rpc_client: RpcClient,
-    ) -> Self {
+    fn new_from_clients(rpc_client: RpcClient, tpu_connection: C) -> Self {
```

Comment:
```suggestion
    fn new_from_client(rpc_client: RpcClient, tpu_connection: C) -> Self {
```

right?  it doesn't take multiple clients as input